src/VisualBasic/CodeCracker/Usage/JsonNetAnalyzer.vb

```diff
@@ -0,0 +1,71 @@
+﻿Imports System.Collections.Immutable
+Imports System.Reflection
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Usage
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Public Class JsonNetAnalyzer
+        Inherits DiagnosticAnalyzer
+
+        Friend Const Title = "Your JSON syntax is wrong."
+        Friend Const MessageFormat = "{0}"
+        Private Const Description = "This diagnostic checks the Json string and triggers if the parsing fails by throwning an exception"
+
+        Friend Shared Rule As New DiagnosticDescriptor(
+            DiagnosticId.JsonNet.ToDiagnosticId(),
+            Title,
+            MessageFormat,
+            SupportedCategories.Usage,
+            DiagnosticSeverity.Error,
+            isEnabledByDefault:=True,
+            description:=Description,
+            helpLink:=HelpLink.ForDiagnostic(DiagnosticId.JsonNet))
+
+        Public Overrides ReadOnly Property SupportedDiagnostics As ImmutableArray(Of DiagnosticDescriptor)
+            Get
+                Return ImmutableArray.Create(Rule)
+            End Get
+        End Property
+
+        Public Overrides Sub Initialize(context As AnalysisContext)
+            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "DeserializeObject", "Public Shared Overloads Function DeserializeObject(Of T)(value As String) As T"), SyntaxKind.InvocationExpression)
+            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Public Shared Overloads Function Parse(json As String) As Newtonsoft.Json.Linq.JObject"), SyntaxKind.InvocationExpression)
+            'context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Nwetonsoft.Json.Linq.JArray.Parse(string)"), SyntaxKind.InvocationExpression)
```

Comment:
Sorry, it should have been removed. Both condition 2 and 3 are handled by 2 due to how VB translates the object differently from C# that is passied in the second parameter. I’ve removed it. Let me know if you need a new pull request.






Sent from Windows Mail





From: Giovanni Bassi
Sent: ‎Thursday‎, ‎February‎ ‎19‎, ‎2015 ‎10‎:‎16‎ ‎AM
To: 'code-cracker/code-cracker'
Cc: Jim Wooley





In src/VisualBasic/CodeCracker/Usage/JsonNetAnalyzer.vb:
> +            SupportedCategories.Usage,
> +            DiagnosticSeverity.Error,
> +            isEnabledByDefault:=True,
> +            description:=Description,
> +            helpLink:=HelpLink.ForDiagnostic(DiagnosticId.JsonNet))
> +
> +        Public Overrides ReadOnly Property SupportedDiagnostics As ImmutableArray(Of DiagnosticDescriptor)
> +            Get
> +                Return ImmutableArray.Create(Rule)
> +            End Get
> +        End Property
> +
> +        Public Overrides Sub Initialize(context As AnalysisContext)
> +            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "DeserializeObject", "Public Shared Overloads Function DeserializeObject(Of T)(value As String) As T"), SyntaxKind.InvocationExpression)
> +            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Public Shared Overloads Function Parse(json As String) As Newtonsoft.Json.Linq.JObject"), SyntaxKind.InvocationExpression)
> +            'context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Nwetonsoft.Json.Linq.JArray.Parse(string)"), SyntaxKind.InvocationExpression)


Why is this commented?

—
Reply to this email directly or view it on GitHub.